OpenRA.Mods.Common/Traits/SupportPowers/SupportPowerManager.cs

```diff
@@ -239,6 +239,9 @@ public void Activate(Order order)
 			if (power == null)
 				return;
 
+			var pos = power.Self.World.Map.CenterOfCell(order.TargetLocation);
+			power = Instances.Where(i => !InstanceDisabled(i)).MinByOrDefault(a => (a.Self.CenterPosition - pos).HorizontalLengthSquared);
```

Comment:
Can you please combine this with / replace the initial assignment just above?  This currently completely overwrites it.